Release 23.10.0 (#13261)
conda/conda: CHANGELOG.md

[//]: # (current developments)

## 23.10.0 (2023-10-30)

> ## ✨ Special announcement ✨

Author: ```suggestion
### ✨ Special announcement ✨
```

This is an announcement about an important change in conda's functionality:

> ### **With this 23.10.0 release we are changing the default solver of conda to [`conda-libmamba-solver`](https://conda.github.io/conda-libmamba-solver/)!** 🥳 🚀

Author: ```suggestion
#### **With this 23.10.0 release we are changing the default solver of conda to [`conda-libmamba-solver`](https://conda.github.io/conda-libmamba-solver/)!** 🥳 🚀
```

The previously "classic" solver is based on [pycosat](https://github.com/conda/pycosat)/[Picosat](http://fmv.jku.at/picosat/) and will remain part of conda for the foreseeable future, a fallback is possible and available.

> ### Why are we switching the solver?

Author: ```suggestion
#### Why are we switching the solver?
```

In short: to make conda faster and more accurate.

A "solver" is the core component of most package managers; it calculates which dependencies (and which version of those dependencies) to install when a user requests to install a package from a package repository. To address growth-related challenges within the conda ecosystem, the conda maintainers, alongside partners Anaconda, Quansight and QuantStack, introduced [a new conda dependency solver based on the Mamba project](https://mamba.readthedocs.io) in December 2022.

> Since July 2023, that [`conda-libmamba-solver`](https://github.com/conda/conda-libmamba-solver) plugin has been included in and automatically installed with all major conda ecosystem installers (miniforge, miniconda, mambaforge and Anaconda Distribution), _with the default solver configuration unchanged_.

Reviewer: Might be me but this could be read like the miniconda distribution and friends will not switch the default solver in the future either? What about adding a follow up sentence that emphasizes that starting with conda 23.10, this won't be the case anymore?

> Since July 2023, that [`conda-libmamba-solver`](https://github.com/conda/conda-libmamba-solver) plugin has been included in and automatically installed with all major conda ecosystem installers (miniforge, miniconda, mambaforge and Anaconda Distribution), _with the default solver configuration unchanged_.

Author: ```suggestion
Since July 2023, the [`conda-libmamba-solver`](https://github.com/conda/conda-libmamba-solver) plugin has been included in all major conda ecosystem installers (miniforge, miniconda, mambaforge and Anaconda Distribution), but was disabled by default. As soon as these installers are updated to contain conda 23.10.0 or later, they will automatically default to using conda-libmamba-solver plugin.
```

Author: @jaimergp How about that?

Reviewer: Perfect!

> ### How can I opt-out?

Author: ```suggestion
#### How can I opt-out?
```

Use one of the following options to configure conda to use the "classic" solver:

- Pass the command line option `--solver=classic` to your conda calls **_or_**
- Set the environment variable `CONDA_SOLVER=classic` **_or_**
- Use the conda configuration system: `conda config --set solver classic`

> ### Where should I file bug reports or feature requests?

Author: ```suggestion
#### Where should I file bug reports or feature requests?
```

Please [head over to the _separate_ conda-libmamba-solver repository](https://github.com/conda/conda-libmamba-solver/issues/new/choose) to file bug reports or feature requests.

> ### Where can I learn more about conda-libmamba-solver?

Author: ```suggestion
#### Where can I learn more about conda-libmamba-solver?
```
